Makefile

```diff
@@ -93,9 +98,17 @@ image: ## Build and push docker image to $registry
 
 .PHONY: deploy 
 deploy: ## Deploy cloud-api-adaptor using the operator, according to install/overlays/$(CLOUD_PROVIDER)/kustomization.yaml file.
+ifneq ($(CLOUD_PROVIDER),)
 	kubectl apply -f install/yamls/deploy.yaml
 	kubectl apply -k install/overlays/$(CLOUD_PROVIDER)
+else
+	$(error CLOUD_PROVIDER is not set)
+endif
 
 .PHONY: delete
 delete: ## Delete cloud-api-adaptor using the operator, according to install/overlays/$(CLOUD_PROVIDER)/kustomization.yaml file.
```

Comment:
same, maybe also worth changing to undeploy?

Reply:
I'll probably address it in a subsequent PR as there are places like e2e where it's used.